src/helpers/ConfigHelper.ts

```diff
@@ -317,29 +437,16 @@ export class ConfigHelper {
       pageViews: SERVER_CONFIG_DEFAULTS_PROMPT_DELAYS.pageViews,
       timeDelay: SERVER_CONFIG_DEFAULTS_PROMPT_DELAYS.timeDelay
     };
-    let categories: Categories | undefined = undefined;
-    if (staticPrompts.slidedown.categories) {
-      categories = TagUtils.limitCategoriesToMaxCount(staticPrompts.slidedown.categories, MAX_CATEGORIES);
-    }
 
-    const slidedown = {
-      enabled: staticPrompts.slidedown.enabled,
-      // for backwards compatibility if not specifically false, then assume true for autoPrompt on slidedown
-      autoPrompt: staticPrompts.slidedown.enabled &&
-        staticPrompts.slidedown.autoPrompt !== false,
-      pageViews: Utils.getValueOrDefault(staticPrompts.slidedown.pageViews,
-          SERVER_CONFIG_DEFAULTS_PROMPT_DELAYS.pageViews),
-      timeDelay: Utils.getValueOrDefault(staticPrompts.slidedown.timeDelay,
-          SERVER_CONFIG_DEFAULTS_PROMPT_DELAYS.timeDelay),
-      actionMessage: staticPrompts.slidedown.actionMessage,
-      acceptButtonText: staticPrompts.slidedown.acceptButton,
-      cancelButtonText: staticPrompts.slidedown.cancelButton,
-      categories,
-    };
+    // cast slidedown to `SlidedownOptions` type
+    const { prompts } = staticPrompts.slidedown as SlidedownOptions;
```

Comment:
@jkasten2 @geekvikas ,
It looks like we don't even look at `prompts` and only look at `staticPrompts`. This is an issue because existing sites don't work in Vikas' latest changes:

```
__jp0({
    "success": true,
    "app_id": "8e7fe838-fbcd-4152-980d-32565a2dcf03",
    "version": 2,
    "features": {
        "restrict_origin": {
            "enable": true
        },
        "cookie_sync": {
            "enable": false
        },
        "metrics": {
            "enable": true,
            "mixpanel_reporting_token": "7c2582e45a6ecf1501aa3ca7887f3673"
        },
        "enable_on_session": false,
        "web_on_focus_enabled": true,
        "session_threshold": 30
    },
    "config": {
        "prompts": {
            "slidedown": {
                "enabled": true,
                "pageViews": 1,
                "timeDelay": 0,
                "autoPrompt": true,
                "categories": {
                    "tags": [{
                        "tag": "key",
                        "label": "Key"
                    }],
                    "updateMessage": "Update your push notification subscription preferences.",
                    "negativeUpdateButton": "Cancel",
                    "positiveUpdateButton": "Save Preferences"
                },
                "acceptButton": null,
                "cancelButton": null,
                "actionMessage": null,
                "isTagsEnabled": true,
                "customizeTextEnabled": false,
                "prompts": [{
                    "text": {
                        "acceptButton": null,
                        "cancelButton": null,
                        "actionMessage": null
                    },
                    "type": "push",
                    "delay": {
                        "pageViews": 1,
                        "timeDelay": 0
                    },
                    "autoPrompt": true,
                    "icon": null
                }, {
                    "categories": [{
                        "tag": "key",
                        "label": "Key"
                    }],
                    "text": {
                        "acceptButton": null,
                        "cancelButton": null,
                        "actionMessage": null,
                        "updateMessage": "Update your push notification subscription preferences.",
                        "negativeUpdateButton": "Cancel",
                        "positiveUpdateButton": "Save Preferences"
                    },
                    "type": "category",
                    "delay": {
                        "pageViews": 1,
                        "timeDelay": 0
                    },
                    "autoPrompt": true,
                    "icon": null
                }]
            }
        },
        "siteInfo": {
            "name": "Web SDK Sandbox",
            "origin": "https://localhost:4001",
            "proxyOrigin": null,
            "defaultIconUrl": null,
            "proxyOriginEnabled": false
        },
        "webhooks": {
            "enable": false,
            "corsEnable": false,
            "notificationClickedHook": null,
            "notificationDismissedHook": null,
            "notificationDisplayedHook": null
        },
        "integration": {
            "kind": "typical"
        },
        "serviceWorker": {
            "path": "/",
            "workerName": "OneSignalSDKWorker.js",
            "registrationScope": "/",
            "updaterWorkerName": "OneSignalSDKUpdaterWorker.js",
            "customizationEnabled": false
        },
        "setupBehavior": {
            "allowLocalhostAsSecureOrigin": false
        },
        "autoResubscribe": true,
        "welcomeNotification": {
            "url": "https://localhost:4001/?_osp=do_not_open",
            "title": null,
            "enable": true,
            "message": null,
            "urlEnabled": false
        },
        "notificationBehavior": {
            "click": {
                "match": "exact",
                "action": "navigate"
            },
            "display": {
                "persist": true
            }
        },
        "staticPrompts": {
            "slidedown": {
                "enabled": true,
                "pageViews": 1,
                "timeDelay": 0,
                "autoPrompt": true,
                "categories": {
                    "tags": [{
                        "tag": "key",
                        "label": "Key"
                    }],
                    "updateMessage": "Update your push notification subscription preferences.",
                    "negativeUpdateButton": "Cancel",
                    "positiveUpdateButton": "Save Preferences"
                },
                "acceptButton": "Allow",
                "cancelButton": "No Thanks",
                "actionMessage": "We'd like to send you notifications for the latest news and updates.",
                "isTagsEnabled": true,
                "customizeTextEnabled": false,
                "prompts": []
            },
            "native": {
                "enabled": false,
                "autoPrompt": true
            },
            "bell": {
                "enabled": false,
                "size": "large",
                "color": {
                    "main": "#E12D30",
                    "accent": "white"
                },
                "dialog": {
                    "main": {
                        "title": "Manage Notifications",
                        "subscribeButton": "Subscribe",
                        "unsubscribeButton": "Unsubscribe"
                    },
                    "blocked": {
                        "title": "Unblock Notifications",
                        "message": "Click here to learn how to unblock notifications."
                    }
                },
                "offset": {
                    "left": 15,
                    "right": 15,
                    "bottom": 15
                },
                "message": {
                    "subscribing": "Thanks for subscribing!",
                    "unsubscribing": "You won't receive notifications again"
                },
                "tooltip": {
                    "blocked": "You've blocked notifications",
                    "subscribed": "You're subscribed to notifications",
                    "unsubscribed": "Subscribe to notifications"
                },
                "location": "bottom-right",
                "hideWhenSubscribed": false,
                "customizeTextEnabled": false
            },
            "fullscreen": {
                "enabled": false,
                "title": "example.com",
                "caption": "You can unsubscribe anytime",
                "message": "This is an example notification message.",
                "acceptButton": "Continue",
                "cancelButton": "No Thanks",
                "actionMessage": "We'd like to send you notifications for the latest news and updates.",
                "autoAcceptTitle": "Click Allow",
                "customizeTextEnabled": false
            },
            "customlink": {
                "enabled": false,
                "style": "button",
                "size": "medium",
                "color": {
                    "button": "#e54b4d",
                    "text": "#ffffff"
                },
                "text": {
                    "subscribe": "Subscribe to push notifications",
                    "unsubscribe": "Unsubscribe from push notifications",
                    "explanation": ""
                },
                "unsubscribeEnabled": true
            }
        },
        "vapid_public_key": "BBfahmiEVx8T_zfNBQD3LutQMGtzN4f0TxZdtvXkbRB-hwDOrhlKrYwlNrc49HE5fHEQnLULgY5rVJCrdK4KleM",
        "onesignal_vapid_public_key": "BMzCIzYqtgz2Bx7S6aPVK6lDWets7kGm-pgo2H4RixFikUaNIoPqjPBBOEWMAfeFjuT9mAvbe-lckGi6vvNEiW0",
        "safari_web_id": "web.onesignal.auto.2cd5950f-b486-4c84-8005-0f30059d0b0c",
        "origin": "https://localhost:4001",
        "http_use_onesignal_com": true,
        "outcomes": {
            "direct": {
                "enabled": false
            },
            "indirect": {
                "enabled": false,
                "notification_attribution": {
                    "minutes_since_displayed": 60,
                    "limit": 10
                }
            },
            "unattributed": {
                "enabled": false
            }
        }
    },
    "generated_at": 1611953476
})
```

Notice two issues here:
1) we have `config.prompts.slidedown.prompts` with the v2 schema being passed in while `config.staticPrompts.slidedown.prompts` is an empty array.
2) There is some faulty logic on the backend resulting in the prompts array be of length 2 (i.e: "push" and "category" type) even though this app was configured with "category" prompt type only.